relayer/src/chain/cosmos.rs

```diff
@@ -930,16 +939,69 @@ impl CosmosSdkChain {
             revision_height: u64::from(status.sync_info.latest_block_height),
         })
     }
-}
 
-fn empty_event_present(events: &[IbcEvent]) -> bool {
-    events.iter().any(|ev| matches!(ev, IbcEvent::Empty(_)))
+    /// Queries the node for both IBC and Cosmos-specific events
+    /// matching a tx hash.
+    fn query_extended_events_for_tx(
+        &self,
+        tx: QueryTxHash,
+    ) -> Result<(Vec<IbcEvent>, Vec<CosmosEvent>), Error> {
+        crate::time!("query_cosmos_events_for_tx");
+        crate::telemetry!(query, self.id(), "query_cosmos_events_for_tx");
+
+        let response = self
+            .block_on(self.rpc_client.tx_search(
+                tx_hash_query(&tx),
+                false,
+                1,
+                1, // get only the first Tx matching the query
+                Order::Ascending,
+            ))
+            .map_err(|e| Error::rpc(self.config.rpc_addr.clone(), e))?;
+
+        match response.txs.into_iter().next() {
+            None => Ok((vec![], vec![])),
+            Some(tx) => {
+                let ibc_events = all_ibc_events_from_tx_search_response(self.id(), &tx);
+                let cosmos_events = all_cosmos_events_from_tx_search_response(self.id(), &tx);
+                Ok((ibc_events, cosmos_events))
+            }
+        }
+    }
+
+    /// A Cosmos-specific variant of `send_messages_and_wait_commit`,
+    /// expecting a Cosmos event in response to the message committed as a
+    /// transaction.
+    pub fn send_message_and_wait_commit_cosmos(
```

Comment:
Do I understand correctly that the only difference between this method and `send_messages_and_wait_commit` is that the former accepts a single `msg: Any`, whereas the latter accepts as parameter a `msgs: TrackedMsgs` (which it splits across multiple tx-es).

The comment and method name here implies that is some more significant difference to the other `send_messages_and_wait_*` method.

Reply:
I did not want to replicate the message batching code, since the only current use of this method sends a single message.
Maybe this needs to be clarified in the documentation.